pkgs/build-support/node/fetch-npm-deps/default.nix

```diff
@@ -208,6 +209,11 @@
 
       forceGitDeps_ = lib.optionalAttrs forceGitDeps { FORCE_GIT_DEPS = true; };
       forceEmptyCache_ = lib.optionalAttrs forceEmptyCache { FORCE_EMPTY_CACHE = true; };
+      gitDepsLockfiles_ = lib.mapAttrs' (
+        name: value: lib.nameValuePair ("NIX_NODEJS_BUILDNPMPACKAGE_LOCKFILE_" + name) value
+      ) gitDepsLockfiles;
+
+      args = lib.filterAttrs (name: value: name != "gitDepsLockfiles") args;
```

Comment:
This line creates the infinite recursion, as nix considers both args to be the same here.
What you want to do is instead replace the `args` at line 219 by `(lib.removeAttrs args [ "gitDepsLockfiles" ])`

Reply:
Oops that variable was originally named `args_`, but I made a typo when rebasing on top of master. Thanks!